tools/fp20compiler/main.cpp

```diff
@@ -84,6 +146,11 @@ int main(int argc, char** argv) {
     fread(buffer, size, 1, fh);
     buffer[size] = '\0';
 
+    if (strchr(buffer, '\r') != NULL) {
+        fprintf(stderr, "found unsupported line-endings\n");
+        exit(1);
+    }
```

Comment:
The commit "fp20compiler: Reject input with unsupported line-endings" triggers on Windows CI: https://travis-ci.org/XboxDev/nxdk/jobs/601952923

This implies that cgc on Windows introduces `\r\n` line-endings.

However, if I manually create a file with those line-endings on Linux, then fp20compiler will get stuck waiting for input via `stdin` (which should never happen). I assume that Windows would do the same (why doesn't it?).

Additionally, the code added in "fp20compiler: Annotate shader sections in output" does not support those line-endings.
As I have no meaningful way to test on Windows at the moment, I'll have to remove that feature from this PR.

I'll split this PR into 2 PRs so only "fp20compiler: Refactor translation function" remains for now (which should be able to handle `\r\n` just fine).
I'll separately PR the other commits, once I'm able to have them working on all platforms.